src/main/scala/org/constellation/consensus/ResolutionService.scala

```diff
@@ -0,0 +1,109 @@
+package org.constellation.consensus
+
+import java.util.concurrent.TimeUnit
+
+import akka.actor.ActorRef
+import akka.util.Timeout
+import org.constellation.Data
+import org.constellation.primitives.{APIBroadcast, Schema}
+import org.constellation.primitives.Schema.{CheckpointBlock, CheckpointCacheData, SignedObservationEdgeCache}
+
+import scala.annotation.tailrec
+import scala.concurrent.{ExecutionContext, Future}
+
+object ResolutionService {
+  implicit val timeout: Timeout = Timeout(5, TimeUnit.SECONDS)
+
+  case class ResolutionStatus(
+                               cb: CheckpointCacheData,
+                               resolvedParents: Seq[(String, Option[SignedObservationEdgeCache])],
```

Comment:
If they are resolved, should these still be Option[...]?

Reply:
Doesn't need to be, they should all be nonEmpty